test/functional/test_framework/wallet.py

```diff
@@ -301,7 +307,10 @@ def getnewdestination(address_type='bech32'):
     elif address_type == 'bech32':
         scriptpubkey = key_to_p2wpkh_script(pubkey)
         address = key_to_p2wpkh(pubkey)
-    # TODO: also support bech32m (need to generate x-only-pubkey)
+    elif address_type == 'bech32m':
+        pubkey = taproot_construct(compute_xonly_pubkey(key.get_bytes())[0]).output_pubkey
+        scriptpubkey = output_key_to_p2tr_script(pubkey)
```

Comment:
Does `taproot_construct()` already return the `scriptPubKey` in `TaprootInfo`?

Reply:
@w0xlt: Yeah it appears to me it does too. Would the `output_key_to_p2tr_script` helper be used elsewhere in future (when `taproot_construct()` isn't used) or should it be ditched?

Reply:
Good point, changed in the latest force-push to use `scriptPubKey` from the returned `TaprootInfo` object. I think it still makes sense to keep the `output_key_to_p2tr_script` helper, first for consistency reasons (we usually have both the `x_to_P2yz` and `x_to_P2yz_script` helpers available) and on the other hand for cases where we might don't use `taproot_construct()` in the future (for bare outputs where we don't know use the script tree or internal key, see e.g. https://github.com/bitcoin/bitcoin/pull/23480).